bindings/twilio/sendgrid/sendgrid.go

```diff
@@ -102,25 +106,43 @@ func (sg *SendGrid) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*b
 	// We allow two possible sources of the properties we need,
 	// the component metadata or request metadata, request takes priority if present
 
+	// Set the email from name, this is optional
+	fromName := ""
```

Comment:
Should this `if` block be nested inside the one checking for the email address?

That is, something like:

	 if sg.metadata.EmailFrom != "" {
	 	 fromName := ""
	 	 if sg.metadata.EmailFromName != "" {
	 	 	 fromName = sg.metadata.EmailFromName
	 	 }
 		fromAddress = mail.NewEmail(fromName, sg.metadata.EmailFrom)
 	}
 	if req.Metadata["emailFrom"] != "" {
	 	  fromName := ""
	 	 if req.Metadata["emailFromName"] != "" {
	 	 	 fromName = eq.Metadata["emailFromName"]
	 	 }
 		fromAddress = mail.NewEmail(fromName, req.Metadata["emailFrom"])
 	}

Otherwise, you could end up in a situation where the email address is set in the component’s metadata, and the name in the request metadata.

Reply:
@ItalyPaleAle I made the suggested change.

Reply:
Thank you!